integration_test/integrate_test.go

```diff
@@ -139,3 +170,82 @@ func assertStringHelper(t *testing.T, name string, subject string, want string,
 		t.Errorf("%s expect %s to be %s but get %s", name, subject, want, get)
 	}
 }
+func assertByteHelper(t *testing.T, name string, subject string, want []byte, get []byte) {
+	if !bytes.Equal(want, get) {
+		t.Errorf("%s expect %s to be %s but get %s", name, subject, want, get)
+	}
+}
+
+type Golden struct {
+	update bool
+}
+
+func (g *Golden) SetUpdate(update bool) {
+	g.update = update
+}
+func (g *Golden) Assert(t *testing.T, actualData []byte) {
```

Comment:
Although it is named `Assert`, but it seems to hold extra update logic here rather than merely assert test result. It is confusing.

Btw, I used [testify](github.com/stretchr/testify/assert) package in promotion's unit test. Its assert.Equal is very easy to use.

Reply:
Yeah, the naming is quite confusing. The reason is that I don't want the user to decide whether to update in every test, the Assert function is kinda like a proxy. Maybe I can add one more abstract layer of function. Do you think `AssertOrUpdate` is a good name for the proxy function?

Reply:
Well, changing the name could help. I could not think of better solution right now.